packages/@lwc/template-compiler/src/codegen/index.ts

```diff
@@ -527,6 +540,11 @@ function transform(codeGen: CodeGen): t.Expression {
             data.push(t.property(t.identifier('svg'), t.literal(true)));
         }
 
+        if (addSanitizationHook) {
```

Comment:
This will generate a `import { renderer } from 'lwc';` statement in the compiled output. This should be the signal to LWS to detect that a custom renderer can be injected for sanitization.